src/scripting/script_initializer.cpp

```diff
@@ -289,6 +289,104 @@ bool
            asOFFSET(BiomeCompoundData, dissolved)) < 0) {
         ANGELSCRIPT_REGISTERFAIL;
     }
+
+    if(engine->RegisterObjectType("ChunkData", 0, asOBJ_REF | asOBJ_NOCOUNT) <
+        0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    ANGELSCRIPT_ASSUMED_SIZE_T;
+    if(engine->RegisterObjectMethod("Biome",
+           "const ChunkData& getChunk(uint64 type) const",
+           asMETHOD(Biome, getChunk), asCALL_THISCALL) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    ANGELSCRIPT_ASSUMED_SIZE_T;
+    if(engine->RegisterObjectMethod("Biome",
+           "array<uint64>@ getChunkKeys() const", asMETHOD(Biome, getChunkKeys),
+           asCALL_THISCALL) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    if(engine->RegisterObjectProperty(
+           "ChunkData", "string name", asOFFSET(ChunkData, name)) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    if(engine->RegisterObjectProperty(
+           "ChunkData", "double density", asOFFSET(ChunkData, density)) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    if(engine->RegisterObjectProperty(
+           "ChunkData", "bool dissolves", asOFFSET(ChunkData, dissolves)) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    if(engine->RegisterObjectProperty(
+           "ChunkData", "uint radius", asOFFSET(ChunkData, radius)) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    if(engine->RegisterObjectProperty(
+           "ChunkData", "uint mass", asOFFSET(ChunkData, mass)) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    if(engine->RegisterObjectProperty(
+           "ChunkData", "uint size", asOFFSET(ChunkData, size)) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    if(engine->RegisterObjectProperty("ChunkData", "double ventAmount",
+           asOFFSET(ChunkData, ventAmount)) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    if(engine->RegisterObjectType(
+           "ChunkCompoundData", 0, asOBJ_REF | asOBJ_NOCOUNT) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    ANGELSCRIPT_ASSUMED_SIZE_T;
+    if(engine->RegisterObjectMethod("ChunkData",
+           "const ChunkCompoundData& getCompound(uint64 type) const",
+           asMETHOD(ChunkData, getCompound), asCALL_THISCALL) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    ANGELSCRIPT_ASSUMED_SIZE_T;
+    if(engine->RegisterObjectMethod("ChunkData",
+           "array<uint64>@ getCompoundKeys() const",
+           asMETHOD(ChunkData, getCompoundKeys), asCALL_THISCALL) < 0) {
+        ANGELSCRIPT_REGISTERFAIL;
+    }
+
+    ANGELSCRIPT_ASSUMED_SIZE_T;
```

Comment:
getMeshListSize actually returns `int` so maybe the return value of that function should be changed to size_t and also getMesh should then take a size_t. This code doesn't work on all possible C++ platforms as AS int is always 32 bits but that isn't required in C++.

Reply:
Okay, I’ll make all those changes, add the ability to scale chunks to the new system, and add the ability to make them damage things and then we will see what happens from there.

Reply:
Okay this is now done